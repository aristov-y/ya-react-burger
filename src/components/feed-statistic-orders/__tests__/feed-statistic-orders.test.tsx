import React from 'react';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import FeedStatisticOrders from '../index';

jest.mock('../feed-statistic-orders.module.css', () => ({
  'OrdersList': 'OrdersList',
  'OrdersCompleted': 'OrdersCompleted'
}))

describe('FeedStatisticOrders', function () {
  it('should render', function () {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<FeedStatisticOrders
        title={"Title"}
        orders={[2,5]}
      />);
    });

    expect(component.toJSON()).toMatchSnapshot();

    act(() => {
      component.update(<FeedStatisticOrders
        title={"Title"}
        orders={[3,4]}
        completed
      />)
    });

    expect(component.toJSON()).toMatchSnapshot();
  });
});
