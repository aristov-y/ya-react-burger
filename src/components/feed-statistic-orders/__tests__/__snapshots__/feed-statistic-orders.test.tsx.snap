// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedStatisticOrders should render 1`] = `
<div>
  <span
    className="text text_type_main-medium"
  >
    Title
  </span>
  <div
    className="OrdersList"
  >
    <span
      className="text text_type_digits-default"
    >
      000002
    </span>
    <span
      className="text text_type_digits-default"
    >
      000005
    </span>
  </div>
</div>
`;

exports[`FeedStatisticOrders should render 2`] = `
<div>
  <span
    className="text text_type_main-medium"
  >
    Title
  </span>
  <div
    className="OrdersList OrdersCompleted"
  >
    <span
      className="text text_type_digits-default"
    >
      000003
    </span>
    <span
      className="text text_type_digits-default"
    >
      000004
    </span>
  </div>
</div>
`;
